src/javascript/app/pages/cashier/cashier.js

```diff
@@ -84,43 +84,42 @@ const Cashier = (() => {
         $(top_up_id).parent().setVisibility(1);
     };
 
-    const showCurrentCurrency = (currency, statement, mt5_logins) => {
-        const has_no_mt5          = mt5_logins.length === 0;
-        const has_no_transaction  = (statement.count === 0 && statement.transactions.length === 0);
-        const el_acc_currency     = getElementById('account_currency');
-        const el_currency_image   = getElementById('account_currency_img');
-        const el_current_currency = getElementById('account_currency_current');
-        const el_current_hint     = getElementById('account_currency_hint');
-        const upgrade_info        = Client.getUpgradeInfo();
-        const can_change          = Client.canChangeCurrency(statement, mt5_logins);
-        const has_upgrade         = upgrade_info.can_upgrade || upgrade_info.can_open_multi
-                                    || can_change;
-        const account_action_text = has_upgrade ? `<br />${localize('[_1]Manage your accounts[_2]', [`<a href=${Url.urlFor('user/accounts')}>`, '</a>'])}` : '';
-        const is_iom_client       = Client.get('residence') === 'im' || State.getResponse('website_status.clients_country') === 'im';
-        const change_text_for_iom = is_iom_client ? localize('time') : localize('time or create an MT5 account');
-        const url_user_account    = `<a href=${Url.urlFor('user/accounts')}>`;
-
-        const missingCriteria = (has_mt5, has_transaction) => {
-            const existing_mt5_msg          = localize('You can no longer change the currency because you\'ve created an MT5 account.') + account_action_text;
-            const existing_transaction_msg  = localize('You can no longer change the currency because you\'ve made a first-time deposit.') + account_action_text;
-
-            return has_mt5 && !has_transaction ? existing_mt5_msg : existing_transaction_msg;
-        };
+    const showCurrentCurrency = (currency, statement, mt5_logins, dxtrade_accounts_list) => {
+        const has_no_real_mt5_or_dxtrade = (!mt5_logins || mt5_logins.filter((account) => account.account_type !== 'demo').length === 0) &&
+            (!dxtrade_accounts_list || dxtrade_accounts_list.filter(account => account.account_type !== 'demo').length === 0);
```

Comment:
Suggestion: You can created a shared function out of this and reuse it in `client-base.js` as well.

Reply:
done ✅  )